src/Sparrow/Platform/Posix/SmapsReader.cs

```diff
@@ -7,6 +7,81 @@
 
 namespace Sparrow.Platform.Posix
 {
+    public class SmapsReaderResults
+    {
+        public string ResultString;
+        public long Size;
+        public long Rss;
+        public long SharedClean;
+        public long SharedDirty;
+        public long PrivateClean;
+        public long PrivateDirty;
+    }
+    
+    public interface ISmapsReaderResultAction
+    {
+        void Add(SmapsReaderResults results);
+    }
+
+    public struct SmapsReaderJsonResults : ISmapsReaderResultAction
+    {
+        private DynamicJsonArray _dja;
+
+        public void Add(SmapsReaderResults results)
+        {
+            var djv = new DynamicJsonValue
+            {
+                ["File"] = results.ResultString,
+                ["Size"] = Sizes.Humane(results.Size),
+                ["Rss"] = Sizes.Humane(results.Rss),
+                ["SharedClean"] = Sizes.Humane(results.SharedClean),
+                ["SharedDirty"] = Sizes.Humane(results.SharedDirty),
+                ["PrivateClean"] = Sizes.Humane(results.PrivateClean),
+                ["PrivateDirty"] = Sizes.Humane(results.PrivateDirty),
+                ["TotalClean"] = results.SharedClean + results.PrivateClean,
+                ["TotalCleanHumanly"] = Sizes.Humane(results.SharedClean + results.PrivateClean),
+                ["TotalDirty"] = results.SharedDirty + results.PrivateDirty,
+                ["TotalDirtyHumanly"] = Sizes.Humane(results.SharedDirty + results.PrivateDirty)
+            };
+            if (_dja == null)
+                _dja = new DynamicJsonArray();
+            _dja.Add(djv);
+        }
+
+        public DynamicJsonArray ReturnResults()
+        {
+            return _dja;
+        }
+    }
+    
+    public struct SmapsReaderNoAllocResults : ISmapsReaderResultAction
+    {
+        private SmapsReaderResults[] _results;
```

Comment:
Use `List<T>` here, instead. No need to write such code ourselfs.

Reply:
Thought about lowering allocations to minimum

Reply:
You aren't, you still do `resize` in the end, and you don't need to, and this is how `List` already work

Reply:
and we do not use these results anyhow.  maybe just throw not implemented ?

Reply:
why throw? Why do _anything_ ? Just have empty method.